internal/cmd/scraper/scraper.go

```diff
@@ -116,12 +116,12 @@ func RunWithEmitters(cfg *Config, emitters []integration.Emitter) error {
 		return fmt.Errorf("you need to configure at least one valid emitter")
 	}
 
-	selfRetriever, err := endpoints.SelfRetriever()
+	selfRetriever, err := endpoints.SelfRetriever(cfg.HostID)
```

Comment:
I would add the replacement mechanism only in the infra_sdk_emitter but not in the endpoint packages that affects the standalone function of POMI. I don't think this is a feature we want to have in k8s since there should be not localhost targetsNames AFAIK

Reply:
Moved to the emitter